libs/core/algorithms/include/hpx/parallel/algorithms/uninitialized_relocate.hpp

```diff
@@ -5,6 +5,253 @@
+    ///                     destination range (deduced).
+    ///                     This iterator type must meet the requirements of a
+    ///                     forward iterator.
+    ///
+    /// \param first        Refers to the beginning of the sequence of elements
+    ///                     the algorithm will be applied to.
+    /// \param count        Refers to the number of elements starting at
+    ///                     \a first the algorithm will be applied to.                   
+    /// \param dest         Refers to the beginning of the destination range.
+    ///
+    /// The assignments in the parallel \a uninitialized_relocate_n algorithm 
+    /// invoked without an execution policy object will execute in sequential 
+    /// order in the calling thread.
+    ///
+    /// \returns  The \a uninitialized_relocate_n algorithm returns \a FwdIter.
+    ///           The \a uninitialized_relocate_n algorithm returns the output
+    ///           iterator to the element in the destination range, one past
+    ///           the last element relocated.
+    ///
+    template <typename InIter, typename Size, typename FwdIter>
+    FwdIter uninitialized_relocate_n(InIter first, Size count, FwdIter dest);
+
+    /// Relocates the elements in the range, defined by [first, last), to an
+    /// uninitialized memory area beginning at \a dest. If an exception is
+    /// thrown during the move-construction of an element, all elements left
+    /// in the input range, as well as all objects already constructed in the
+    /// destination range are destroyed. After this algorithm completes, the
+    /// source range should be freed or reused without destroying the objects.
+    ///
+    /// \note   Complexity: time: O(n), space: O(1)
+    ///         1)  For "trivially relocatable" underlying types (T) and
+    ///             a contiguous iterator range [first, first+count):
+    ///             `count*sizeof(T)` bytes are copied.
+    ///         2)  For "trivially relocatable" underlying types (T) and
+    ///             a non-contiguous iterator range [first, first+count):
+    ///             `count` memory copies of sizeof(T) bytes each are performed.
+    ///         3)  For "non-trivially relocatable" underlying types (T):
+    ///             `count` move assignments and destructions are performed.
+    ///
+    /// \note   Declare a type as "trivially relocatable" using the
+    ///         `HPX_DECLARE_TRIVIALLY_RELOCATABLE` macros found in
+    ///         <hpx/type_support/is_trivially_relocatable.hpp>.
+    ///
+    /// \tparam ExPolicy    The type of the execution policy to use (deduced).
+    ///                     It describes the manner in which the execution
+    ///                     of the algorithm may be parallelized and the manner
+    ///                     in which it executes the assignments.
+    /// \tparam InIter      The type of the source iterator first (deduced).
```

Comment:
I think this should be FwdIter1 (at least the arguments below have FwdIter1 and FwdIter2)